### DISCUSSION TITLE
Foundry Code Tool

### DISCUSSION THREAD

**Discussion by suddenly1990** [Cyfrin/foundry-full-course-cu#2701] [Ideas]
Because I wanted to avoid repetitive tasks, I created a tool for code checking. You’re welcome to use it, and I would appreciate any feedback or suggestions.

[FoundryFileHelper](https://github.com/suddenly1990/foundryfilehelper), is a powerful tool aimed at enhancing the efficiency of developers working with Foundry by automating key tasks. It offers the following features:

1.  **Automatic File Generation**: When creating a new contract in the src folder, corresponding test files and deployment scripts are automatically generated in the test and script directories, ensuring consistent project structure.

2. **File Deletion Assistance**: If you delete a contract from the src folder, the tool will prompt whether to delete the associated test and script files, helping maintain a clean project.

3. **Generate Makefile**: The tool provides the Generate Foundry Makefile function, which creates a Makefile template in the project directory, simplifying project automation and build processes.

4. **Generate Environment Variables**: With Generate Foundry Env, you can generate an .env file that includes critical fields for easier configuration in your projects.
```
such as:
	•	SEPOLIA_RPC_URL=
	•	PRIVATE_KEY=
	•	PRIVATE_KEY= 
```

This plugin is perfect for developers aiming to streamline their workflow and focus more on development. For more details and to try it out, visit the [GitHub repository](https://github.com/suddenly1990/foundryfilehelper).

You can now find it on the VSCode Marketplace, but there seems to be an issue with the icon for some reason. If you encounter any other issues or have suggestions, feel free to reach out!

**Comment by suddenly1990** [comment]
The next development goal for the FoundryFileHelper plugin is quite exciting, as it involves integrating AI-driven technology to automate unit test code generation. Here’s a breakdown of the upcoming features:

1. **WriteAgent**: This agent will handle the initial creation of unit test code, leveraging AI to write effective test cases based on the project’s structure and requirements.

2. **CheckAgent**: After the code is generated, CheckAgent will analyze it to identify any issues, ensuring that the code is free of errors or vulnerabilities.

3. **EvaluationAgent**: This final agent will ensure that the test code meets all necessary standards and is ready for deployment, providing a comprehensive review to finalize the testing process.

This AI-powered extension will further streamline the development process by minimizing manual testing efforts, which is a huge boost in efficiency. Also, the FoundryFileHelper can already be found on the VSCode Marketplace, allowing developers to easily access and integrate it into their workflow.

This approach will make Foundry projects more robust and efficient.

**Reply by EngrPips** [reply]
This is mind blowing.
